src/herder/simulation/SimulationTxSetFrame.cpp

```diff
@@ -75,12 +76,23 @@ SimulationTxSetFrame::sortForApply()
     res.reserve(mTransactions.size());
 
     auto resultIter = mResults.cbegin();
+    uint32_t count = 0;
     for (auto const& txEnv : mTransactions)
     {
         res.emplace_back(SimulationTransactionFrame::makeTransactionFromWire(
-            mNetworkID, txEnv, resultIter->result));
+            mNetworkID, txEnv, resultIter->result, count));
         ++resultIter;
+        // Transaction generation guarantees that the number of transactions in
```

Comment:
actually that's not exactly the reason: see my comment on how we generate the stream (and how we need a comment that explains what's going on, ie round robin over partitions).